Add interacting-with-sap guide (#396)
wso2/ballerina-integrator: docs/content/src/learn/tutorials/saas-integrations/sap/interacting-with-sap/README.md

You can find other integration modules from [wso2-ballerina](https://github.com/wso2-ballerina) GitHub organization.

Following are the sections available in this guide.

- [What you'll build](#what-youll-build)
- [Prerequisites](#prerequisites)
- [Implementation](#implementation)
- [Testing](#testing)

## What you'll build

This application demonstrates a scenario where a client application interacts with a SAP endpoint using BAPI and IDoc. The BAPI request would provide the response from the backend and if an error happens during the message flow, it would be returned to the user. A successful IDoc message send would return the associated transaction id.

## Prerequisites

- [Java](https://www.oracle.com/technetwork/java/index.html)
- Ballerina Integrator
- A Text Editor or an IDE
    > **Tip**: For a better development experience, install the `Ballerina Integrator` extension in [VSCode](https://code.visualstudio.com/).
- SAP R/3 instance with `sapidoc3.jar`, `sapjco3.jar` and the native SAP JCo library.

## Implementation

> If you want to skip the basics, you can download the git repo and directly move to the `Testing` section by skipping the `Implementation` section.

> 1. Create a new project.

Reviewer: Check these working properly in the site.

> **Windows**: Copy the Windows native SAP jcolibrary `sapjco3.dll` to `<WINDOWS_HOME>/system32`.

4. Create a `lib` directory in the project root and copy the `sapidoc3.jar` and `sapjco3.jar` .jar files into it. Point these .jar files in the `Ballerina.toml` file as follows.
```
[platform]
target = "java8"

    [[platform.libraries]]
    module = "sapjco"
    path = "lib/sapjco3.jar"

    [[platform.libraries]]
    module = "sapidoc"
    path = "lib/sapidoc3.jar"
```
5. Add a `ballerina.conf` file and create .bal files with meaningful names as shown in the project structure given below.
```
interacting-with-sap
├── Ballerina.toml
├── ballerina.conf
└── src
    └── sap_producer
        ├── Module.md
        ├── sap_producer.bal
        ├── resources
        └── tests
            ├── resources
            └── sap_producer_test.bal
```

6. Add SAP endpoint properties in the `ballerina.conf` file. Given below is a sample with mandatory properties.
```
DESTINATION_NAME = "CPT"
SAP_CLIENT = "800"
USERNAME = "wso2_user"
PASSWORD = "wso2_pass"
ASHOST = "/H/217.116.29.154/S/3299/H/10.100.5.120/S/3200"
SYSNR = "01"
LANGUAGE = "en"
```

> 7. A SAP client could be configured as follows. Ballerina Integrator VS Code plugin contains a snippet for the same, which could be loaded using the autocomplete feature for `client/sap` keyword.

Reviewer 2: ```suggestion
7. A SAP client can be configured as follows. Ballerina Integrator VS Code plugin contains a snippet for the same, which could be loaded using the autocomplete feature for `client/sap` keyword.
```

```
sap:ProducerConfig producerConfigs = {
    destinationName: config:getAsString("DESTINATION_NAME"),
    sapclient: config:getAsString("SAP_CLIENT"),
    username: config:getAsString("USERNAME"),
    password: config:getAsString("PASSWORD"),
    ashost: config:getAsString("ASHOST"),
    sysnr: config:getAsString("SYSNR"),
    language: config:getAsString("LANGUAGE")
};

sap:Producer sapProducer = new (producerConfigs);
```

> 8. Functions `sendBapi()` and `sendIdoc()` could be used to send the BAPI requests and IDoc messages respectively.

Reviewer 2: ```suggestion
8. Functions `sendBapi()` and `sendIdoc()` can be used to send the BAPI requests and IDoc messages respectively.
```

## Testing

Let’s build the module. Navigate to the project directory and execute the following command.

```bash
ballerina build sap_producer
```

> The build command would create an executable .jar file. Now run the .jar file created in the above step. Path to the `ballerina.conf` could be provided using the `--b7a.config.file` option.

Reviewer 2: ```suggestion
The build command creates an executable .jar file. Now run the .jar file created in the above step. Path to the `ballerina.conf` could be provided using the `--b7a.config.file` option.
```
